// cli-plugin-eslint/template/.eslintrc.js
const resolve = require('./config/webpack.config.js').resolve;
module.exports = {
  env: {
    browser: true,
    commonjs: true,
    es6: true
  },
  extends: ['eslint-config-airbnb', 'prettier', 'plugin:jsx-a11y/strict'],
  parserOptions: {
    ecmaFeatures: {
      jsx: true
    },
    ecmaVersion: 8,
    sourceType: 'module'
  },
  plugins: ['react', 'jsx-a11y', 'prettier'],
  rules: {
    indent: 'off',
    'react/jsx-filename-extension': [1, { extensions: ['.js', '.jsx'] }],
    'comma-dangle': ['error', 'never'],
    'arrow-parens': ['error', 'as-needed'],
    'no-plusplus': 'off',
    'no-underscore-dangle': [0],
    'class-methods-use-this': 'off',
    'import/no-extraneous-dependencies': 'off',
    'react/forbid-prop-types': 'off',
    'global-require': 'off',
    semi: ['error', 'always'],
    'prettier/prettier': 'error',
    'jsx-a11y/click-events-have-key-events': 'warn',
    'jsx-a11y/anchor-is-valid': [ 'error', {
      'components': [ 'Link' ],
      'specialLink': [ 'to' ]
    }],
    'no-debugger': 'warn',
    'linebreak-style': 0
  },
  root: true,
  settings: {
    'import/resolver': {
      'eslint-import-resolver-webpack-alias': { resolve }
    }
  }
};
